src/nvim/ex_cmds.c

```diff
@@ -5563,7 +5566,12 @@ static buf_T *show_sub(exarg_T *eap, pos_T old_cusr,
   linenr_T highest_num_line = 0;
   int col_width = 0;
 
+  // if we fail to split the window, we don't want to modify orig_buf
+  bool split_success = false;
+
   if (split && win_split((int)p_cwh, WSP_BOT) != FAIL) {
+    split_success = true;
```

Comment:
`preview_buf` is set in this same block, do we need both? Could we just NULL out `preview_buf` if we want it to be skipped?

Reply:
Yes, I think this should work